tower-breaker/src/lib.rs

```diff
@@ -0,0 +1,103 @@
+//! A middleware that blocks readiness based on an inner service's load.
+
+pub mod error;
+
+use futures::{future, try_ready, Future, Poll};
+use tower_load::Load;
+use tower_service as svc;
+
+/// Determmines readiness based on an `M`-typed load metric.
+pub trait Breaker<M> {
+    type Error;
+
+    /// Checks to see whether the breaker is open, given a load metric.
+    ///
+    /// If the breaker is closed, NotReady is returned and the task ust be
+    /// notified when the breaker should be polled again with an updated load
+    /// metric.
+    fn poll_breaker(&mut self, load: M) -> Poll<(), Self::Error>;
```

Comment:
If the `load` value changes between calls to `poll_breaker`, then I assume the inner state needs to be updated to reflect the last value of `load` as the most recent? In the past, arguments to `poll_*` fns have been problematic because how to handle the argument changing between calls was not obvious.